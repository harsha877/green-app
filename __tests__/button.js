import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import CustumeButton from '../src/components/button'

test("Button Component Loading",()=>{
    render(<CustumeButton name="welcome" onPressHandler={()=>{}} type="button" colorsuper = "#1ca3ec"/>)
 
 })
 
 test("Button component snapshot",()=>{
    let ButtonComponent=render(<CustumeButton name="welcome" onPressHandler={()=>{}} type="button" colorsuper = "#1ca3ec"/>).toJSON();
    expect(ButtonComponent).toMatchSnapshot();
 })

describe("Testing Button component",()=>{
    let ButtonComponent;
    beforeEach(()=>{
    ButtonComponent=render(<CustumeButton name="welcome" onPressHandler={()=>{}} type="button"  colorsuper = "#1ca3ec"/>);
    
    })
    
    test("Testing Styles",()=>{
        const props=ButtonComponent.toJSON();
        let Styles=props.children[0].props.style;
        expect(Styles).toStrictEqual({
            padding: 0,
            borderRadius: 50,
            backgroundColor: "#1ca3ec",
            paddingVertical: 10,
            paddingHorizontal: 10,
            margin: 10,
            width: 125,
            alignSelf: "center",
        });})
        
        test("Testing Button Text",()=>{
            const {getAllByText}=ButtonComponent;
            expect(getAllByText("welcome").length).toBe(1);})

        test("Testing Styles",()=>{
            const props=ButtonComponent.toJSON();
            let Styles=props.children[0].children[0].props.style;
            expect(Styles).toStrictEqual({
                color: "#ffffff",
                fontSize: 18,
                textAlign: "center",
                padding: 5,
            });})
   
});

describe("Testing Button component with type = list",()=>{
    let ButtonComponent;
    beforeEach(()=>{
    ButtonComponent=render(<CustumeButton name="click" onPressHandler={()=>{}} type="list"  colorsuper = "#1ca3ec"/>);
    
    })
    
    test("Testing Styles",()=>{
        const props=ButtonComponent.toJSON();
        let Styles=props.children[0].props.style;
        expect(Styles).toStrictEqual({
            borderRadius: 20,
            backgroundColor: "#1ca3ec",
            paddingVertical: 10,
            paddingHorizontal: 10,
            margin: 4,
            borderColor: "black",
        });})
        
        test("Testing Button Text",()=>{
            const {getAllByText}=ButtonComponent;
            expect(getAllByText("click").length).toBe(1);})

        test("Testing Styles",()=>{
            const props=ButtonComponent.toJSON();
            let Styles=props.children[0].children[0].props.style;
            expect(Styles).toStrictEqual({
                color: "#ffffff",
                fontSize: 18,
                textAlign: "center",
                padding: 5,
            });})
   
});