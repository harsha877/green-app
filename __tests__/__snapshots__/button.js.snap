// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#1ca3ec",
        "borderRadius": 50,
        "margin": 10,
        "padding": 0,
        "paddingHorizontal": 10,
        "paddingVertical": 10,
        "width": 125,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      welcome
    </Text>
  </View>
</View>
`;
